numpyro/distributions/discrete.py

```diff
@@ -620,6 +623,23 @@ def sample(self, key, sample_shape=()):
     def log_prob(self, value):
         if self._validate_args:
             self._validate_sample(value)
+        value = jax.device_get(value)
+        if (
+            self.is_sparse
+            and not isinstance(value, jax.core.Tracer)
+            and jnp.size(value) > 1
+        ):
+            shape = lax.broadcast_shapes(self.batch_shape, jnp.shape(value))
+            rate = jnp.broadcast_to(self.rate, shape).reshape(-1)
+            nonzero = jnp.broadcast_to(value > 0, shape).reshape(-1).nonzero()
+            value = jnp.broadcast_to(value, shape).reshape(-1)
```

Comment:
nit: I guess we can swap the order of those two lines:
```
value = jnp.broadcast_to(value, shape).reshape(-1)
nonzero = (value > 0).nonzero()
```
Btw, do we need to call `.nonzero()` here?

Reply:
Done. I think `.nonzero()` was required for tests to pass in an earlier commit, but now things pass without it, so I've removed.